src/builder/QuarkBuilderBase.sol

```diff
@@ -0,0 +1,642 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.27;
+
+import {IQuarkWallet} from "quark-core/src/interfaces/IQuarkWallet.sol";
+
+import {Actions} from "src/builder/actions/Actions.sol";
+import {Accounts} from "src/builder/Accounts.sol";
+import {BridgeRoutes} from "src/builder/BridgeRoutes.sol";
+import {EIP712Helper} from "src/builder/EIP712Helper.sol";
+import {Math} from "src/lib/Math.sol";
+import {MorphoInfo} from "src/builder/MorphoInfo.sol";
+import {Strings} from "src/builder/Strings.sol";
+import {PaycallWrapper} from "src/builder/PaycallWrapper.sol";
+import {QuotecallWrapper} from "src/builder/QuotecallWrapper.sol";
+import {PaymentInfo} from "src/builder/PaymentInfo.sol";
+import {TokenWrapper} from "src/builder/TokenWrapper.sol";
+import {QuarkOperationHelper} from "src/builder/QuarkOperationHelper.sol";
+import {List} from "src/builder/List.sol";
+
+contract QuarkBuilderBase {
+    /* ===== Output Types ===== */
+
+    struct BuilderResult {
+        // Version of the builder interface. (Same as VERSION, but attached to the output.)
+        string version;
+        // Array of quark operations to execute to fulfill the client intent
+        IQuarkWallet.QuarkOperation[] quarkOperations;
+        // Array of action context and other metadata corresponding 1:1 with quarkOperations
+        Actions.Action[] actions;
+        // Struct containing containing EIP-712 data for a QuarkOperation or MultiQuarkOperation
+        EIP712Helper.EIP712Data eip712Data;
+        // Client-provided paymentCurrency string that was used to derive token addresses.
+        // Client may re-use this string to construct a request that simulates the transaction.
+        string paymentCurrency;
+    }
+
+    /* ===== Constants ===== */
+
+    string constant VERSION = "0.1.2";
+
+    /* ===== Custom Errors ===== */
+
+    error AssetPositionNotFound();
+    error FundsUnavailable(string assetSymbol, uint256 requiredAmount, uint256 actualAmount);
+    error InvalidActionChain();
+    error InvalidActionType();
+    error InvalidInput();
+    error MaxCostTooHigh();
+    error MissingWrapperCounterpart();
+    error InvalidRepayActionContext();
+
+    struct ActionIntent {
+        address actor;
+        uint256[] amountOuts;
+        string[] assetSymbolOuts;
+        uint256[] amountIns;
+        string[] assetSymbolIns;
```

Comment:
Can you add some documentation here? It seems like you intend for `amountOut` to mean the assets that a user sends out and `amountIn` to mean the assets that a user gets back. It's a tad confusing because it's the inverse of what `amountOut` and `amountIn` mean in Uniswap.

Reply:
Yeah haha. amountIn and amountOut for Uniswap contract is from Uniswap's perspective 😄 
In a sense it's kind of like written in the same approach as uniswap lol. 

but, sure will document this here. 👍